command/check.go

```diff
@@ -0,0 +1,130 @@
+package command
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	HealthCritical = 2
+	HealthWarn     = 1
+	HealthPass     = 0
+	HealthUnknown  = 3
+)
+
+type AgentCheckCommand struct {
+	Meta
+}
+
+func (c *AgentCheckCommand) Help() string {
+	helpText := `
+Usage: nomad check
+  
+  Display state of the Nomad agent. The exit code of the command is Nagios
+  compatible and could be used with alerting systems.
+
+General Options:
+
+  ` + generalOptionsUsage() + `
+
+Agent Check Options:
+  
+  -min-peers
+     Minimum number of peers that a server is expected to know.
+
+  -min-servers
+     Minumum number of servers that a client is expected to know.
+`
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *AgentCheckCommand) Synopsis() string {
+	return "Displays health of the local Nomad agent"
+}
+
+func (c *AgentCheckCommand) Run(args []string) int {
+	var minPeers, minServers int
+
+	flags := c.Meta.FlagSet("check", FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	flags.IntVar(&minPeers, "min-peers", 0, "")
+	flags.IntVar(&minServers, "min-servers", 1, "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	client, err := c.Meta.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("error initializing client: %s", err))
+		return HealthCritical
+	}
+
+	info, err := client.Agent().Self()
+	if err != nil {
+		c.Ui.Output(fmt.Sprintf("unable to query agent info: %v", err))
+		return HealthCritical
+	}
+	if _, ok := info["stats"]["nomad"]; ok {
```

Comment:
could this panic if `info["stats"]` doesn't return an initialized map. Probably want two layers of if okay checks